completed Project

2bleO opened the pull request:
**In this project I:**

- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Apply React best practices and language style guides in code.
- Use store, actions and reducers in React.

2bleO committed:
Create React app

2bleO committed:
Install redux toolkit && react-redux

2bleO committed:
setup linters

2bleO committed:
Setup file structure

2bleO committed:
fix linter errors

2bleO committed:
Merge pull request #1 from 2bleO/initialize

Initialize

2bleO committed:
Install redux devtools extension

2bleO committed:
install react-bootstrap

2bleO committed:
add redux && redux-thunk

2bleO committed:
install axios

2bleO committed:
install react router dom

2bleO committed:
install redux-logger

2bleO committed:
install prop-types

2bleO committed:
install react-router-bootstrap

2bleO committed:
install Font Awsome

2bleO committed:
Configure store

2bleO committed:
Create action and reducer

2bleO committed:
Implement provider and router

2bleO committed:
Setup path and styling

2bleO committed:
Fix style lint missing new line end of file

2bleO committed:
Merge pull request #2 from 2bleO/store-and-reducers

Store and reducers

2bleO committed:
Create Home component

2bleO committed:
create details component

2bleO committed:
create filter component

2bleO committed:
home page header image

2bleO committed:
Merge pull request #3 from 2bleO/components

Components

2bleO committed:
Add Tests

2bleO committed:
Fix Linter errors

2bleO committed:
Merge pull request #4 from 2bleO/tests

Add Tests

2bleO committed:
Modify Readme

2bleO committed:
Get coins when app load

Test to see if this way netlify can fetch data.

2bleO committed:
Add nomics attribution

2bleO committed:
Merge pull request #5 from 2bleO/final-touches

Final touches

2bleO committed:
Add explanatory video

2bleO committed:
Merge pull request #6 from 2bleO/final-touches

Add explanatory video

oloomoses commented on src/components/Home.jsx:
```diff
@@ -0,0 +1,84 @@
+/* eslint-disable max-len */
+import React, { useState } from 'react';
```
- [x] Kindly do not disable linter because it enforces code quality. Instead, you should fix the issues raised.

peterrobert replied:
This change has been implemented well

oloomoses commented on src/components/Home.jsx:
```diff
@@ -0,0 +1,84 @@
+import { Col, Row } from 'react-bootstrap';
+import { LinkContainer } from 'react-router-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faArrowAltCircleRight } from '@fortawesome/free-solid-svg-icons';
+// import { getCoins } from '../redux/cryptodata/cryptodata';
+import Filter from './Filter';
```
- [x] Kindly delete all the commented lines of code because they make your work look messy.

peterrobert replied:
This change has been implemented well

oloomoses commented on src/redux/configureStore.js:
```diff
@@ -0,0 +1,14 @@
+
+import reducer from './cryptodata/cryptodata';
+
+const store = createStore(
+  reducer,
+  composeWithDevTools(applyMiddleware(thunk, logger)),
+);
```
- [x] Please do not log redux data on the browser console as it may cause site security issues. Kindly clear the logged data displayed on the browser console.

peterrobert replied:
This change has been implemented well

oloomoses reviewed (changes requested):
Hi @2bleO, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] Kindly add unit tests for the components and functions that you have created.

_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

2bleO committed:
changes

delete all the commented lines of code
do not disable linter
do not log redux data on the browser

2bleO committed:
Add more tests

2bleO committed:
Fix linter errors

peterrobert commented on src/components/Home.test.js:
```diff
@@ -0,0 +1,43 @@
+import { BrowserRouter as Router } from 'react-router-dom';
+import { createStore, applyMiddleware } from 'redux';
+import { Provider } from 'react-redux';
+// import { store } from '../redux/configureStore';
```
### [OPTIONAL]

- [ ] It will be great to delete commented code. This will keep your code clean. Remember, you can always use git history to go back to the previous version of your code.

peterrobert reviewed (approved):
Hi @2bleO, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
